src/compiler/checker.ts

```diff
@@ -39195,7 +39198,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // TS 1.0 spec (April 2014): 8.3.2
             // Constructors of classes with no extends clause may not contain super calls, whereas
             // constructors of derived classes must contain at least one super call somewhere in their function body.
-            const containingClassDecl = node.parent as ClassDeclaration;

```

Comment:
this cast was "unsafe" and actually redundant, `node.parent` is already types as `ClassLikeDeclaration`